### ISSUE TITLE
Add a confirmation before sending a cancellation email to a customer

### ISSUE THREAD

**Issue by RachL** [openfoodfoundation/openfoodnetwork#8477]
## Description
**- As a:** enterprise user or super admin
**- On page:** `/admin/orders/RXXXX/edit` OR `/admin/orders/bulk_management`
**- I want to be able to do:** If I delete the last item and a modal is shown to tell me the order will be canceled and I say yes to it, a second modal should appear and ask me if I want the customer to have a notification email.

Context : In CA in particular, it is often the case that orders are re-ajusted / merged in a way that one order has to be canceled. Currently customers receive automatically an email informing about the cancellation.

CA would like this email to remain an option, so customers are not worried in cases of merging orders.

https://openfoodnetwork.slack.com/archives/C018S101R4J/p1631933446034700

## Acceptance Criteria & Tests

_Scenario 1 as an enterprise user_

1. Log in and go to `/admin/orders?q[s]=completed_at+desc`
2. Edit an order and delete all items
3. A modal appear telling you the order will be cancelled. Proceed. 
4. A second modal shows up and ask if you wish the cancellation email to be sent to the customer. Say yes.
5. Check that the cancellation email (current behavior) is still correctly sent.

_Scenario 2 as an enterprise user_

1. Log in and go to `/admin/orders?q[s]=completed_at+desc`
2. Edit an order and delete all items
3. A modal appear telling you the order will be cancelled. Proceed. 
4. A second modal shows up and ask if you wish the cancellation email to be sent to the customer. Say no.
5. Check that the cancellation email is not sent to the customer.

_Scenario 3 & 4_

Check that scenario 1 & 2 work also for the super admin.

_Scenario 5, 6, 7 & 8_

Repeat scenario 1 to 4 but instead of deleting items from `/admin/orders/RXXXX/edit` delete them from `/admin/orders/bulk_management`.

**Comment by zatteo** [comment]
Hi everyone,

I started to work on the BOM page and will try to make a proposition at the end of the week.

**Comment by jibees** [comment]
@RachL 
Currently deleting the last item of an order, is not possible due to this popup:

<img width="529" alt="Capture d’écran 2022-01-19 à 10 16 47" src="https://user-images.githubusercontent.com/296452/150100912-58968121-70dd-4e9a-a216-b9ed7c5a460f.png">

If I understood correctly, we should remplace this popup by the popup that ask the user if they want to send the cancellation email or not (but cancel the order in any case). Right?

**Comment by RachL** [comment]
@jibees no all current popup needs to stay. You need to add a third one. Happy to discuss if there are other solutions than a popup though

**Comment by jibees** [comment]
I read the last comment of Jana, that said:

> We suggested adding a check box "Send out cancellation e-mail to the user" which is checked by default to the current "Are you sure you want to cancel this order"-Modal?

That's why I started with a modal like this, appearing when deleting the last item of an order:
<img width="516" alt="Capture d’écran 2022-01-19 à 11 39 39" src="https://user-images.githubusercontent.com/296452/150114355-34cd2a6f-f215-4417-899b-c57022b904df.png">

But then I realized you wrote:
> a second modal should appear and ask me if I want the customer to have a notification email.

So, I agree: let's do a quick chat together, and discuss about modals (or not).

**Comment by RachL** [comment]
@jibees I'm fine with your suggested modal: 

![image](https://user-images.githubusercontent.com/6525576/150115359-75cffe86-2213-40e1-a748-f158af76b014.png)

Go for it :)

**Comment by jibees** [comment]
_Context: Bulk Order Management_

It seems that, currently, there is no popup that prevent the consumer to not delete the last item of an order. I'm then wondering if we should consider the _Bulk Order Management_ context anyway (and then add more code to check if the item that the admin delete is the last of the current order) or skip this case for this papercut?

**Comment by RachL** [comment]
@jibees it's handle in : https://github.com/openfoodfoundation/openfoodnetwork/pull/8294 you need to work on top of this PR

**Comment by jibees** [comment]
~I don't how to work on top of a PR that is actually blocked by mine, I feel like we are in a deadlock ;)~

_Conclusion_: the linked PR will only handle the `/orders/[ORDER_NUMBER]/edit` page. After the two PRs merged (#8294 and #8766 then), I will open to handle such a case in the BOM interface.

**Comment by jibees** [comment]
Should we open another issue (linked to this one) to handle the "cancel order + send email cancellation" on the BOM interface?
